lib/ansible/modules/net_tools/basics/get_url.py

```diff
@@ -461,11 +461,21 @@ def main():
                 checksum_url = checksum
                 # download checksum file to checksum_tmpsrc
                 checksum_tmpsrc, checksum_info = url_get(module, checksum_url, dest, use_proxy, last_mod_time, force, timeout, headers, tmp_dest)
-                lines = [line.rstrip('\n') for line in open(checksum_tmpsrc)]
-                os.remove(checksum_tmpsrc)
+                with open(checksum_tmpsrc) as f:
+                    lines = [line.rstrip('\n') for line in f]
                 lines = dict(s.split(None, 1) for s in lines)
                 filename = url_filename(url)
-                [checksum] = (k for (k, v) in lines.items() if v.strip('./') == filename)
+
+                # Look through each line in the checksum file for a hash corresponding to
+                # the filename in the url, returning the first hash that is found.
+                for cksum in (s for (s, f) in lines.items() if f.strip('./') == filename):
+                    checksum = cksum
+                    break
+                else:
+                    checksum = None
```

Comment:
I'm unsure if it would be better to put `checksum = None` before the loop and just remove this `else` statement entirely, or if this is ok.